paasta_tools/utils.py

```diff
@@ -1204,14 +1207,26 @@ def get_git_url(service: str, soa_dir: str = DEFAULT_SOA_DIR) -> str:
     url will be git@github.yelpcorp.com:services/test.
 
     :param service: The service name to get a URL for
+    :param soa_dir: Path to soa configs directory
     :returns: A git url to the service's repository"""
     general_config = service_configuration_lib.read_service_configuration(
         service, soa_dir=soa_dir
     )
-    default_location = "git@github.yelpcorp.com:services/%s" % service
+    system_config = load_system_paasta_config()
```

Comment:
not very happy about calling this method every time in so many places, it is quite expensive. maybe allow passing `system_configs` as an argument?

Reply:
I decided to slim down the PR to only replace sysgit usages, specifically when we trigger deploys, and when we update autotune things. I did add a todo to move other usages like for `github.yelpcorp.com` into the system config file. We can do it at a later time, but for now I need to time box.